public/app/features/explore/ResponseErrorContainer.test.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { configureStore } from '../../store/configureStore';
+import { ResponseErrorContainer } from './ResponseErrorContainer';
+import { Provider } from 'react-redux';
+import { render, screen } from '@testing-library/react';
+import { ExploreId } from '../../types';
+import { DataQueryError, LoadingState } from '@grafana/data';
+
+describe('ResponseErrorContainer', () => {
+  it('shows error message if it does not contain refId', async () => {
+    setup({
+      message: 'test error',
+    });
+    screen.getByText('test error');
```

Comment:
just a nit here and in the subsequent tests, it should make failing tests more clear and the code bit more explicit, as this relies only on the fact that `getByText` throws.

```suggestion
    expect(screen.queryByText('test error')).not.toBeInTheDocument();
```

Reply:
Not that I would mind changing this but does that mean nobody uses `getByText`? Because `expect(screen.getByText('test error')).not.toBeInTheDocument();` will throw before it goes into the expect function. Also regarding the `toBeInTheDocument` does it do anything more than a `not.isNull`? Personally I like when I need to remember as little as possible regarding APIs :D

Reply:
`toBeInTheDocument` also performs some additional checks that may or may not be relevant (in this case probably not), like expecting the element to actually be in the document root
for example
```
expect(true).not.isNull;
```
will pass, while
```
expect(true).toBeInTheDocument();
```

but yeah, probably in this test is quite superfluous.

Reply:
also, i guess the main advantage is when you are not querying by text but by using something `queryByRole`, so you don't actually check for the text to be present, but for an element with a specific meaning to be in the document, which also probably doesn't apply here

Reply:
Yeah did not think about `expect(true).not.isNull;` that you can accidentally do something like that.